src/Common/Core/Impl/Threading/IMainThread.cs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.Common.Core.Threading {
+    public interface IMainThread {
```

Comment:
Well in Windows there is always 'main thread' as long as there is a window - which is almost always true. 

---
In reply to: [81223138](https://github.com/Microsoft/RTVS/pull/2408#discussion_r81223138) [](ancestors = 81223138,81221364)